README.md

```diff
@@ -1,66 +1,125 @@
-[foundry](https://github.com/foundry-rs/foundry) that can be used to create mock Solidity
-contracts able to mock variables and external calls.
+[![Version](https://img.shields.io/npm/v/@defi-wonderland/foundry-mock-generator?label=Version)](https://www.npmjs.com/package/@defi-wonderland/foundry-mock-generator)
+[![License: MIT](https://img.shields.io/badge/License-MIT-blue.svg)](https://github.com/defi-wonderland/prophet-sdk/blob/main/LICENSE)
 
-# Features
+# Foundry Mock Generator
+A plugin for [foundry](https://github.com/foundry-rs/foundry) that automatically generates Solidity mocks for every contract in your project.
+
+## Features
 
 - Get rid of your folder of "mock" contracts and **just use
   foundry**.
-- Keep your tests **simple** with a easy mock functions.
-- Mock up external calls and variables in a beautiful and orderly fashion.
+- Keep your tests **simple** with straightforward mock functions.
+- Mock up external calls and internal variables in a beautiful and orderly fashion.
 
 ## Installation
 
-You can install the tool via yarn:
+You can install the plugin via yarn:
 
 ```bash
-yarn add foundry-mock-generator@<latest-canary>
+yarn add @defi-wonderland/foundry-mock-generator --save-dev
 ```
 
 ## Basic Usage
 
+### Creating mocks
+
 To generate the mock contracts all you have to do is run:
 
 ```bash
-yarn foundry-mock-generator --contracts <path/to/contracts> --out <path/to/foundry/out> --genDir <path/to/generate/contracts>
+yarn foundry-mock-generator --contracts path/to/contracts
 ```
 
-Note: --contracts required by the user, tha path to the solidity contracts to mock
-Note: --genDir default path is `solidity/test/mock-contracts`, the path to generate the mock contracts
-Note: --out default path is `out`, the path that has foundry's compiled artifacts
+The `foundry-mock-generator` command accepts the following options:
+
+Option      | Default                           | Notes
+------------|-----------------------------------|-------
+`contracts` | —                                 | The path to the solidity contracts to mock
+`out`       | `./out`                           | The path that has the compiled artifacts
+`genDir`    | `./solidity/test/mock-contracts`  | The path to the generated mock contracts
+`ignore`    | []                                | A list of directories to ignore
+
+### Using mocks
 
-1. To use the mock contracts in your tests just import them
+Let's say you have a Greeter contract in your project:
 
-```JavaScript
-import { MockMyContractName } from '/path/to/mock-contracts/MockMyContractName.sol'
+```solidity
+contract Greeter {
+  string internal _greeting;
+
+  constructor(string memory greeting) {
+    _greeting = greeting;
+  }
+
+  function greet() public view returns (string memory) {
+    return _greeting;
+  }
+}
 ```
 
-2. Create the mock contract and allow cheatcodes for it
+After running the generator, you will have a mock contract located at `${genDir}/MockGreeter.sol`:
```

Comment:
its true only if `MockGreeter.sol` right inside of `genDir`, otherwise, the path will be `${genDir}/**/MockGreeter.sol`